pyprotista/parsers/ident/comet_2020_01_4_parser.py

```diff
@@ -1,65 +1,135 @@
 """Engine parser."""
-import multiprocessing as mp
-import sys
-from io import BytesIO
 
 import numpy as np
 import pandas as pd
 import regex as re
-from loguru import logger
-from lxml import etree
-from tqdm import tqdm
+import xml.etree.cElementTree as etree
 
 from pyprotista.parsers.ident_base_parser import IdentBaseParser
 
+element_tag_prefix = "{http://psidev.info/psi/pi/mzIdentML/1.2}"
 
-def _mp_specs_init(func, reference_dict, mapping_dict):
-    func.reference_dict = reference_dict
-    func.mapping_dict = mapping_dict
 
+def _iterator_xml(xml_file, mapping_dict):
+    version = ""
+    stage = 0
 
-def _get_single_spec_df(spectrum):
-    """Primary method for reading and storing information from a single spectrum.
+    modifications = []
+    sequence = {}
+    peptide_lookup = {}
 
-    Attributes:
-        reference_dict (dict): dict with reference columns to be filled in
-        mapping_dict (dict): mapping of engine level column names to ursgal unified column names
+    spec_results = {}
+    spec_ident_items = []
+    spec_records = []
 
-    Args:
-        spectrum (xml Element): namespace of single spectrum with potentially multiple PSMs
+    modification_mass_map = {}
+    mod_name = ""
+    fixed_mods = {}
 
-    Returns:
-        (pd.DataFrame): dataframe containing spectrum information
+    for event, entry in etree.iterparse(xml_file):
+        entry_tag = entry.tag
 
-    """
-    spectrum = etree.parse(BytesIO(spectrum)).getroot()
-    spec_records = []
-    spec_level_dict = _get_single_spec_df.reference_dict.copy()
-    spec_level_dict["spectrum_id"] = spectrum.attrib["spectrumID"].split("scan=")[-1]
-
-    # Iterate children
-    for psm in spectrum.findall(".//{*}SpectrumIdentificationItem"):
-        psm_level_dict = spec_level_dict.copy()
-        psm_level_dict.update(
-            {
-                _get_single_spec_df.mapping_dict[k]: psm.attrib[k]
-                for k in _get_single_spec_df.mapping_dict
-                if k in psm.attrib
-            }
-        )
-        cv_param_info = {
-            c.attrib["name"]: c.attrib["value"] for c in psm.findall(".//{*}cvParam")
-        }
-        psm_level_dict.update(
-            {
-                _get_single_spec_df.mapping_dict[k]: cv_param_info[k]
-                for k in _get_single_spec_df.mapping_dict
-                if k in cv_param_info
-            }
-        )
+        if entry_tag == (f"{element_tag_prefix}PeptideEvidence"):
```

Comment:
also all of these parentheses are redundant

Reply:
and here (and below) if we do not want to use `element_tag_prefix` then one could write
` if entry_tag.endswith("PeptideEvidence"):`